File: backend/models/job.py
import uuid
import enum
from datetime import datetime
from sqlalchemy import Column, String, DateTime, Enum, ForeignKey, Integer
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.orm import relationship
from models.base import Base

class JobType(str, enum.Enum):
    ppt = "ppt"
    podcast = "podcast"
    video = "video"

class JobStatus(str, enum.Enum):
    queued = "queued"
    processing = "processing"
    completed = "completed"
    failed = "failed"

class Job(Base):
    __tablename__ = "jobs"

    job_id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    type = Column(Enum(JobType), nullable=False)
    status = Column(Enum(JobStatus), default=JobStatus.queued, nullable=False)
    progress = Column(Integer, default=0, nullable=False)
    result = Column(JSONB, nullable=True) # or JSON for standard DB without jsonb
    error = Column(String, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

File: backend/models/Job.js
const mongoose = require("mongoose");

const jobSchema = new mongoose.Schema(
  {
    jobId: { type: String, required: true, unique: true, index: true },
    userId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
    type: {
      type: String,
      enum: ["lecture", "podcast", "flashcards", "upload"],
      required: true,
    },
    sourcePdfUrl: { type: String },
    status: {
      type: String,
      enum: ["pending", "partial", "complete", "failed"],
      default: "pending",
      index: true,
    },
    payload: { type: Object, default: {} },
    output: { type: Object, default: {} },
    error: { type: String, default: null },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Job", jobSchema);
